app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsViewModel.kt

```diff
@@ -133,60 +69,140 @@ class SessionDetailsViewModel @JvmOverloads constructor(
+            val session = repository.loadSelectedSession()
+            val uri = feedbackUrlComposer.getFeedbackUrl(session).toUri()
+            openFeedBack.postValue(uri)
         }
-        R.id.menu_item_share_session_json -> {
-            val formattedSessions = session.toJson()
-            viewActionHandler.shareAsJson(formattedSessions)
-            true
+    }
+
+    fun share() {
+        viewModelScope.launch(executionContext.database) {
+            val session = repository.loadSelectedSession()
+            val timeZoneId = repository.readMeta().timeZoneId
+            simpleSessionFormat.format(session, timeZoneId).let { formattedSession ->
+                shareSimple.postValue(formattedSession)
+            }
         }
-        R.id.menu_item_add_to_calendar -> {
-            viewActionHandler.addToCalendar(session)
-            true
+    }
+
+    fun shareToChaosflix() {
+        viewModelScope.launch(executionContext.database) {
+            val session = repository.loadSelectedSession()
+            jsonSessionFormat.format(session).let { formattedSession ->
+                shareJson.postValue(formattedSession)
+            }
         }
-        R.id.menu_item_flag_as_favorite -> {
-            session.highlight = true // Required: Update property because refreshUI refers to its value!
-            repository.updateHighlight(session)
-            repository.notifyHighlightsChanged()
-            viewActionHandler.refreshUI()
-            true
+    }
+
+    fun addToCalendar() {
+        viewModelScope.launch(executionContext.database) {
+            val session = repository.loadSelectedSession()
+            addToCalendar.postValue(session)
         }
-        R.id.menu_item_unflag_as_favorite -> {
-            session.highlight = false // Required: Update property because refreshUI refers to its value!
-            repository.updateHighlight(session)
-            repository.notifyHighlightsChanged()
-            viewActionHandler.refreshUI()
-            true
+    }
+
+    fun favorSession() {
+        viewModelScope.launch(executionContext.database) {
+            val session = repository.loadSelectedSession()
+            val favoredSession = Session(session).apply {
+                highlight = true // Required: Update property because updateHighlight refers to its value!
+            }
+            repository.updateHighlight(favoredSession)
+            repository.notifyHighlightsChanged() // TODO Remove when FahrplanFragment uses Flow
         }
-        R.id.menu_item_set_alarm -> {
-            viewActionHandler.showAlarmTimePicker()
-            true
+    }
+
+    fun unfavorSession() {
+        viewModelScope.launch(executionContext.database) {
+            val session = repository.loadSelectedSession()
+            val unfavoredSession = Session(session).apply {
+                highlight = false // Required: Update property because updateHighlight refers to its value!
+            }
```

Comment:
This could be dryied with `favorSession`

Reply:
:x: I could not come up with a good function name to be used for the merged code part.